runtime/android/core/src/org/xwalk/core/XWalkUpdater.java

```diff
@@ -92,17 +150,36 @@
          * update or the download (from the specified URL) is cancelled
          */
         public void onXWalkUpdateCancelled();
+
+        public void onXWalkUpdateStarted();
+        public void onXWalkUpdateFailed();
+        public void onXWalkUpdateCompleted();
     }
 
     private static final String XWALK_APK_MARKET_URL = "market://details?id=org.xwalk.core";
-    private static final String TAG = "XWalkActivity";
+
+    private static final String META_XWALK_APK_URL = "xwalk_apk_url";
+    private static final String META_XWALK_SILENT_DOWNLOAD = "xwalk_silent_download";
+
+    private static final String PRIVATE_LIB_DIR = "xwalkcorelib";
+
+    private static final String[] XWALK_LIB_RESOURCES = {
+        "libxwalkcore.so",
+        "icudtl.dat",
+        "xwalk.pak",
+        "classes.dex",
+    };
```

Comment:
make sense. then only check libxwalkcore.so and replace PLATFORM with the real one.